### PR TITLE
Fix Notification issues

### LINKED ISSUE #8182: Missing notifications from forum
I didn't get any notification from any reply of this forum thread I interacted with.

https://friendica.mrpetovan.com/display/7c104781-195e-2f26-ed87-4ca864495840

### LINKED ISSUE #8187: Wrong notification URL
I would love it because now i have some ugly side effects, so that i cant react on contact requests. It throws an error like this

```
#0 [internal function]: Friendica\Module\HTTPException\PageNotFound::content(Array)
#1 /Friendica_Install/htdocs/src/App/Page.php(310): call_user_func(Array, Array)
#2 /Friendica_Install/htdocs/src/App/Page.php(364): Friendica\App\Page->initContent(Object(Friendica\App\Module), Object(Friendica\App\Mode))
#3 /Friendica_Install/htdocs/src/App.php(583): Friendica\App\Page->run(Object(Friendica\App), Object(Friendica\App\BaseURL), Object(Friendica\App\Mode), Object(Friendica\App\Module), Object(Friendica\Core\L10n), Object(Friendica\Core\Config\JitConfig), Object(Friendica\Core\PConfig\JitPConfig))
#4 /Friendica_Install/htdocs/index.php(27): Friendica\App->runFrontend(Object(Friendica\App\Module), Object(Friendica\App\Router), Object(Friendica\Core\PConfig\JitPConfig), Object(Friendica\App\Authentication), Object(Friendica\App\Page))
#5 {main}
```

_Originally posted by @hoergen in https://github.com/friendica/friendica/issues/8181#issuecomment-578983328_

### LINKED ISSUE #8193: DB error 1054
### Observed behavior
I checked my log file and I found ~700 DB errors 1054. Maybe this is the cause of the issue I reported here? https://github.com/friendica/friendica/issues/8180#issuecomment-578854078

### Details of naughty behavior

``` 
2020-01-28T19:24:33Z index [ERROR]: DB Error {"code":1054,"error":"Unknown column 'abort' in 'field list'","callstack":"Database::insert, BaseRepository::insert, Notify::insert, notification, check_item_notification, check_user_notification, Item::insert, Processor::postItem","params":"INSERT INTO `notify` (`name`, `url`, `photo`, `uid`, `iid`, `parent`, `type`, `verb`, `otype`, `date`, `abort`) VALUES ('Steffen K9 \ud83d\udc30', 'https:\/\/libranet.de\/profile\/alfred', 'https:\/\/libranet.de\/photo\/3b2c1a18929163223620833d46bf017c-5.jpg?ts=1580226164', 33, 54291489, 54289238, 128, 'http:\/\/activitystrea.ms\/schema\/1.0\/tag', 'item', '2020-01-28 19:24:33', '')"} - {"file":"Database.php","line":714,"function":"e","uid":"2a9b3f","process_id":4875}
``` 

### Friendica version you encountered the problem
2020.03-dev

### Friendica source (git, zip)
git

### PHP version
7.2

### SQL version
10.1 MariaDB

### PR DESCRIPTION
**Pull request by nupplaphil**
FollowUp #8170 

- Add `name_cache` entry in enotify
- Add "unset()" in notify repository for additional field "abort"
- Fix enotify link generation
- Add new "API" notification object and factory

Tested on my friendica.philipp.info node and:
- Bell counter works again
- New entry for comments in the notification-dropdown (Fixes #8182 - see https://friendica.philipp.info/display/f3ad7b1c-155e-2e99-e325-a84095934917)
- Link in the notification dropdown is working (again)
- Click on the link reduces the counter again (see https://friendica.philipp.info/display/f3ad7b1c-515e-2e75-550e-1b1487841944 )
- Intro ignore is now working again (Fixes #8187 )
- Intro accept is now working again (Addresses https://github.com/friendica/friendica/issues/7999#issuecomment-576007661 )
- Adding new notifications are working again (Fixes #8193)

### CODE REVIEW

**Comment by MrPetovan** [comment]
Wait, why do we need the `abort` field if it isn't supposed to be saved?

**Comment by nupplaphil** [comment]
Because an addon could force an "abort" of this save..

**Comment by MrPetovan** [comment]
Then the fields should be emptied instead. This is wrong.

**Comment by nupplaphil** [comment]
what do you mean? In case abort is true, we return null, in the other case, we remove the "abort" field again..

And in case we empty out the fields, we would save an empty array. So there has to be a check if we want to abort the save ..

**Comment by nupplaphil** [comment]
ok, I changed the behavior.
If the field array is empty, we stop the save :-)

File: src/Model/Notify.php [hunk 1]
```diff
@@ -163,4 +163,29 @@ public static function formatMessage($name, $message)
 
 		return $message;
 	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	protected function mapFields(array $data)
```
Comment (MrPetovan) [inline src/Model/Notify.php]: Do we still need this?

Comment (nupplaphil) [reply src/Model/Notify.php]: yes. The `Model\Notify` contains `msg_html` and `msg_text` for the notification table... and `timestamp` and `relDate`.. That's what I mean with "attributes which aren't db relevant"..

Comment (MrPetovan) [reply src/Model/Notify.php]: Shouldn't these be moved to the Notification entity?

Comment (nupplaphil) [reply src/Model/Notify.php]: Hmmm.. I guess you're right.. I'll add a "Notification" entity at API and add a factory method for it.. Much more clean of course!

**Comment by nupplaphil** [comment]
wow .. little cause big harm! .. too many fields storing in the repo and BUFF everything was gone ^^

**Comment by AlfredSK** [comment]
Remember the add-on PR. :-)

**Comment by MrPetovan** [comment]
Thanks for the reminder!

**Comment by AlfredSK** [comment]
Notifications do work again. But it seems the info bubble on the bell icon is missing again. :)

**Pull request merged**